site/docs/manual/Architecture.md

```diff
@@ -130,35 +173,111 @@ entity.addComponent(ComponentA)
 the engine will try to reuse a `ComponentA` instance, from the pool of components previously created, and it won't allocate a new one instead.
 When releasing that component, by calling `entity.removeComponent(ComponentA)`, it will get returned to the pool, ready to be used by another entity.
 
-ECSY should know how to reset a component to its original state, that's why it's highly recommended that components implements a `reset` method to get the benefits from pooling.
+ECSY should know how to reset a component to its original state, if your component has the proper schema defined, ECSY will do this for you.
+
+#### Custom Components
+
+Sometimes it's not possible to define a component with a schema. If you still want to get the benefits of object pooling you can override some of the methods on the `Component` class.
 
 ```javascript
-// Example of components with `reset` methods implemented
+class ColorArray extends Component {
+  constructor(props) {
+    // Pass false to disable using the schema for default values.
+    super(false);
 
-class List extends Component {
-  constructor() {
+    // Set your own default values instead
     this.value = [];
   }
 
+  copy(src) {
+    this.value.length = src.value.length;
+
+    for (let i = 0; i < src.value.length; i++) {
+      const srcColor = src.value[i];
+      const destColor = this.value[i];
+
+      destColor.r = srcColor.r;
+      destColor.g = srcColor.g;
+      destColor.b = srcColor.b;
+    }
+  }
+
+  /**
+   * We don't need to override clone in this case.
+   * clone() {
+   *  return new this.constructor().copy(this);
+   * }
+   **/
+
   reset() {
-    this.value.length = 0;
+    this.value.forEach(color => {
+      color.r = 0;
+      color.g = 0;
+      color.b = 0;
+    });
   }
 }
+```
 
-class Position extends Component {
-  constructor() {
-    this.reset();
+In extreme cases, you may experience performance bottlenecks due to the default implementation of the `Component` class. If you experience this, you can override that specific component with your own faster implementation.
+
+#### Disable Component Pooling
+
+In other cases you may want to disable component pooling altogether. Some components can't be copied or cloned properly.
+
+In this case you can disable component pooling when you first register a component:
+
+```javascript
+class AudioListener extends Component {
```

Comment:
Probably this is something we would like to define when registering the component instead of the component itself as you can change the behaviour on your app. ?

Reply:
I don't know if we can make that assumption. You could avoid using component pooling for other reasons and still want to use the default implementations of `clone()`, `copy()`, and `reset()`. I can remove the `throw new Error("unimplemented")` code though if that's confusing.